### PR TITLE
Port mson tests

### PR DESCRIPTION
**Pull request by klokane**
Small refactoring for tests:
- include tests for MSON into new testing convetions, 
- remove duplicit code for different wrappers

@danielgtaylor  can you, please, review, it is base on your work

### CODE REVIEW

File: test/test-RefractDataStructureTest.cc [hunk 1]
```diff
@@ -4,187 +4,47 @@ using namespace draftertest;
...
+TEST_MSON("array-typed-content");
+TEST_MSON("resource-nested-member");
+TEST_MSON("array-sample");
+TEST_MSON("object-sample");
+TEST_MSON("enum-members-description");
+TEST_MSON("array-reference");
+TEST_MSON("reference-override");
+TEST_MSON("enum-variants");
+
+#undef TEST_MSON
```
Comment (pksunkara) [inline test/test-RefractDataStructureTest.cc]: What about `#undef TEST_MSON_`. And can we have a better name for `TEST_MSON_`?

File: test/draftertest.h [hunk 2]
```diff
@@ -12,20 +12,25 @@
 #include "SerializeResult.h"
 #include "sosJSON.h"
 
+
+#define TEST_DRAFTER(description, category, name, tag,  wrapper, options, mustBeOk) TEST_CASE(description " " category " " name, "[" tag "][" category "]") { \
+    REQUIRE(FixtureHelper::handleResultJSON(wrapper, "test/fixtures/" category "/" name, options, mustBeOk)); \
+}
```
Comment (pksunkara) [inline test/draftertest.h]: We could have actually had `TEST_MSON` here itself. Why are we treating it specially when compared to other macros?

**Comment by pksunkara** [comment]
Looks good. Just a few things.

File: test/test-RefractDataStructureTest.cc (hunk 1 above)
Comment (klokane) [inline test/test-RefractDataStructureTest.cc]: Oh good catch, there were original just `TEST_MSON`. Macro `TEST_MSON_` I add later to allow failing parser output.

File: test/draftertest.h (hunk 2 above)
Comment (klokane) [inline test/draftertest.h]: I mean, we should do it in same way for other tests.

There is no reason to have specific macro accessible for other test files, it just filling "global namespace"

Comment (pksunkara) [inline test/draftertest.h]: Okay, we can do that later then. Please fix the other comment and I can merge this.

File: test/test-RefractDataStructureTest.cc (hunk 1 above)
Comment (klokane) [inline test/test-RefractDataStructureTest.cc]: If you don't like name, propose some better :)

Comment (pksunkara) [inline test/test-RefractDataStructureTest.cc]: `TEST_MSON_SUCCESS`? Just having an underscore at the end looks unreadable.

Comment (danielgtaylor) [inline test/test-RefractDataStructureTest.cc]: I think `TEST_MSON_SUCCESS` makes sense.

**Comment by danielgtaylor** [comment]
Looks good to me, just that one comment :+1:

**Comment by pksunkara** [comment]
@klokane Let's merge this tomorrow.

**Pull request merged**